timelock-impl/src/main/java/com/palantir/atlasdb/timelock/adjudicate/LeaderElectionDurationAccumulator.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.adjudicate;
+
+import com.palantir.logsafe.Preconditions;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.timelock.feedback.LeaderElectionDuration;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.LongConsumer;
+import org.immutables.value.Value;
+import org.immutables.value.Value.Parameter;
+
+public class LeaderElectionDurationAccumulator {
+    private Set<LeadersContext> alreadyReportedLeaderElections = ConcurrentHashMap.newKeySet();
```

Comment:
Looks like this grows unbounded. Is this a problem / should we consider using some kind of cache with an ExpiryAfterWrite/Access?

Reply:
Let's say that you have 50 clients, so each leader election you store 50 * 2 UUIDs, which are 16 bytes IIRC. So that's 160 bytes per leader election. You'd need say >6500 leader elections to occur in the lifetime of this accumulator before you'd have 1MB of space allocated.

I'm not by any means against making this an expiry cache, but probably won't be necessary.